charts/openstack-cluster/templates/control-plane/kubeadm-control-plane.yaml

```diff
@@ -21,6 +21,22 @@ preKubeadmCommands:
 {{- end }}
 {{- end }}
 
+{{- define "openstack-cluster.controlplane.kubeadmConfigSpec.ignitionKubeProxyConfiguration" -}}
+{{- with .kubeProxyConfiguration }}
+files:
+  - path: /etc/kube-proxy-configuration.yaml
+    content: |
+      ---
+      apiVersion: kubeproxy.config.k8s.io/v1alpha1
+      kind: KubeProxyConfiguration
+      {{- toYaml . | nindent 6 }}
+    owner: root:root
+    permissions: "0644"
+preKubeadmCommands:
+  - cat /etc/kube-proxy-configuration.yaml >> /run/kubeadm.yml
+{{- end }}
+{{- end }}
```

Comment:
Can this be done by just gating the path and command on `.Values.ignitionBasedOS`? I'm not that keen on the duplication.

Reply:
Inside the block I was unable to access the  `.Values.ignitionBasedOS` flag. The helm chart would fail to render because the `.ignitionBasedOS` variable didn't exist. Maybe that happens because it is in the `.kubeProxyConfiguration` context?  I'm not sure. 

Perhaps there's a trick to getting the root level objects